### PR TITLE
chore(web): move lm-worker into its own folder 🎡

### PR DESCRIPTION
**Pull request by mcdurdin**
This starts the reorg required for splitting the lm worker into its own module. Tests are still not working. But builds are going.

Some decisions to be made around the final path for wrapping of the code. I think it may be possible to do something a whole lot more elegant by moving the wrapping process into the KMW build and leaving the LM Worker output as a pure tsc output -- given only KMW wants the wrapped worker anyway (tests aside). The actual wrapping code could still be in the lm-worker folder.

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by jahorton** [comment]
> I think it may be possible to do something a whole lot more elegant by moving the wrapping process into the KMW build and leaving the LM Worker output as a pure tsc output -- given only KMW wants the wrapped worker anyway (tests aside).

> (tests aside)

If nothing else, this should make a lot of difference.

From my perspective, moving the wrapping process into the main KMW build is far _less_ elegant.  I'd rather add an extra package into the overall workspace to facilitate things rather than have it within the main KMW build.

File: common/web/lm-worker/package.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    "karma-edge-launcher": "^0.4.2",
+    "karma-firefox-launcher": "^1.1.0",
+    "karma-fixture": "^0.2.6",
+    "karma-html2js-preprocessor": "^1.1.0",
+    "karma-ie-launcher": "^1.0.0",
+    "karma-json-fixtures-preprocessor": "0.0.6",
+    "karma-mocha": "^2.0.1",
+    "karma-mocha-reporter": "^2.2.5",
+    "karma-safari-launcher": "^1.0.0",
+    "karma-teamcity-reporter": "^1.1.0",
```
Comment (jahorton) [inline common/web/lm-worker/package.json:32]: Feels a little weird to have all of these in-browser testing dependencies here for the _worker-only_ project... but I guess we do have the unit tests that directly test its installation directly into WebWorkers, which would best be tested via an actual browser run.

**Review by jahorton — COMMENTED** [review]
As far as the split of the worker's code into its own package/project, I'm fine with that.

I'm less fine with the way that the worker's code is then integrated where needed into the components that need it, at least with only this PR's changes.  If the differentiated `browser.tsconfig.json` and `tsconfig.json` both still exist, they should be integrating their respective 'installation' of the wrapped worker code.  That's the raison-d'être for their differentiation.

Doesn't surprise me that unit tests aren't working at this point in the PR ladder, as this'll definitely present problems when it comes to testing `@keymanapp/input-processor`, possibly among other components.

**Comment by jahorton** [comment]
After reviewing the full chain... yeah, I see why it's having to be linked the way it is now.  TypeScript throws a fit if you try to directly include it because of all the Worker-space global references that aren't valid in standard document-based web code.

**Review by jahorton — APPROVED** [review]

**Comment by mcdurdin** [comment]
> After reviewing the full chain... yeah, I see why it's having to be linked the way it is now. TypeScript throws a fit if you try to directly include it because of all the Worker-space global references that aren't valid in standard document-based web code.

Many crying and tears here but basically ... yeah.

**Comment by jahorton** [comment]
> > After reviewing the full chain... yeah, I see why it's having to be linked the way it is now. TypeScript throws a fit if you try to directly include it because of all the Worker-space global references that aren't valid in standard document-based web code.
> 
> Many crying and tears here but basically ... yeah.

Well... I decided to do one last round of investigation into it... and I found something.

I think https://github.com/keymanapp/keyman/pull/6644 might bring a little joy.

**Pull request merged**